datalad/support/annexrepo.py

```diff
@@ -297,6 +297,21 @@ def add_remote(self, name, url, options=None):
                              " -S %s" % c.ctrl_path)
             writer.release()
 
+    def set_remote_url(self, name, url, push=False):
+        """Overrides method from GitRepo in order to set
+        remote.<name>.annex-ssh-options in case of a SSH remote."""
+
+        super(AnnexRepo, self).set_remote_url(name, url, push=push)
+        from datalad.support.network import is_ssh
+        if is_ssh(url):
+            c = ssh_manager.get_connection(url)
```

Comment:
ok then you enclose further:
```python
try:
            writer.set_value("remote \"%s\"" % name,
                             "annex-ssh-options",
                             "-o ControlMaster=auto"
                             " -S %s" % c.ctrl_path)
finally:
            writer.release()
```
so if set_value fails (I don't know how and don't want to know if it could or not -- just taking no chances), then writer would release ... although I must say that not sure if that would be a good thing -- does it save at that point (then better just to fail -- which would be your code now), or just removes some lock on the `.git/config` (then my try/final)